sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -0,0 +1,344 @@
+import fs from "fs-extra";
+import nise from "nise";
+import queryString from "query-string";
+import { isBrowser, blobToString } from "./utils";
+import { customConsoleLog } from "./customConsoleLog";
+
+const env = isBrowser() ? (window as any).__env__ : process.env;
+const isRecording = env.TEST_MODE === "record";
+const isPlayingBack = env.TEST_MODE === "playback";
+
+let nock: any;
+if (!isBrowser() && (isRecording || isPlayingBack)) {
+  nock = require("nock");
+}
+
+if (isBrowser() && isRecording) {
+  customConsoleLog();
+}
+
+if (isPlayingBack) {
+  // Providing dummy values to avoid the error
+  env.ACCOUNT_NAME = "fakestorageaccount";
+  env.ACCOUNT_KEY = "aaaaa";
```

Comment:
More work on ENV variables 
https://github.com/Azure/azure-sdk-for-js/issues/4284